### ISSUE TITLE
.col does not wrap by itself

### ISSUE THREAD

**Issue by fogx** [quasarframework/quasar#8464]
quasar grid allows columns to [auto-resize depending on other elements](https://quasar.dev/layout/grid/row#Setting-one-column-width). If the total columns equals 12, the .col element will not wrap by itself, as it is given a size of 0. This leaves the content of that column hanging on the side.

with padding/width>0
![image](https://user-images.githubusercontent.com/10303647/107763700-544aa280-6d2f-11eb-94e8-54da4a39055d.png)
without styling/ width=0
![image](https://user-images.githubusercontent.com/10303647/107763730-5b71b080-6d2f-11eb-9c1f-54ba14ce880e.png)

This is not made clear in the documentation and is a pitfall atleast me and [another user](https://github.com/quasarframework/quasar/issues/5894) have stumbled upon. The issue is fixed by giving the element a min-width of 1px, which will force it to wrap.

I would suggest adding an info box at the variable width section 

(Side note: .col will not have an exact equal width to .col-1, maybe due to rounding errors?)

**Comment by pdanpdan** [comment]
Hello. Col splits the available space in equal chunks. If the available space is 0 then the size is 0. It has nothing to do with col-1.
Do you have an example where your easily makes sense?

**Comment by fogx** [comment]
[here is a codepen to exemplify the issue](https://codepen.io/wKon/pen/abBBMdb)
Setting `min-width` to atleast 1 pixel wraps the column.

The width of .col and .col-1 isn't important for this issue.
It is also visible in the codepen. `col` will be 0.02px smaller than `col-1`.

**Comment by rstoenescu** [comment]
You got .col11 + .col-1. What's an additional .col supposed to do? When you use .col-<number> you specify exactly how much each cell takes up.

You either use .col-<number> or simple .col -- Doesn't makes sense to combine them. Maybe you wanted .col-12?

**Comment by fogx** [comment]
> You got .col11 + .col-1. What's an additional .col supposed to do? When you use .col- you specify exactly how much each cell takes up.
> 
> You either use .col- or simple .col -- Doesn't makes sense to combine them. Maybe you wanted .col-12?

i used `col-11` and `col-1` to exemplify the issue. 

To reiterate, the issue is that it is not clear (to me and atleast one other person) that **`.col` will not wrap by itself without a width**. The grid documentation page does not mention this case, and using the [codepen in the documentation](https://quasar.dev/layout/grid/row#Setting-one-column-width) will not show this error, because the styling applied prevents the width from going to 0. ([Pen to better explain the issue](https://codepen.io/wKon/pen/GRNNLLV))
`grid-grow`, is probably the best class for the desired use-case. However, `grid-grow` is not mentioned on the Grid Row Documentation page, which makes it easy to miss.
So to repeat my suggestion:
I suggest: 
- adding a section in the documentation that shows this pitfall
- mentioning the use of `.col-grow`, because it is not otherwise mentioned on the page.

**Comment by fogx** [comment]
@rstoenescu also kind of rude to close the thread without properly reading the issue. I'm just trying to help improve the project/documentation.

**Comment by rstoenescu** [comment]
The best way to help out is to identify what can be improved in the docs and PRing it.

From my perspective:
1. .col shares the remaining space on a row; in this particular case, 0.
2. There isn't any sense in using .col-<number> + .col and expecting it to wrap, since .col is designed to share the remaining space (point number 1)
3. .col-grow described here: https://quasar.dev/layout/grid/introduction-to-flexbox#Distribution-of-Size

With all due respect, calling the owner of the project "rude" is not gonna get you far. Especially when he pours all his sweat and blood to offer his work for free while working day and night on it. You are thousands while we are just a few. So if you see an opportunity to improve something, kindly asking to PR it. Sometimes there might be minor stuff that can be better explained, but it's not like the docs are written in a hurry, although Quasar's huge.

So bottom line, let's call it a truce and value each other's time. Again, not trying to be a smart-ass, just put yourself in the shoes of opensource developers. If you can help then help and all the community members will thank you, not just the maintainers.

**Comment by fogx** [comment]
I greatly appreciate your work and the time you put into this project. At the same time, i too am trying to help you with your work, and it is quite demotivating when my attempt to help is squashed without properly looking at the issue.

If we find a consensus on how to improve, i will gladly create a PR to fix the issue myself. Imo it makes more sense to talk to you about an issue before creating a PR and getting my suggestion stomped (like it was here). 

Now back to the topic: 
the codepen i posted shows why i think it makes sense to use `.col` after `.col-`.  All other elements, such as col-1 and col-grow will go to the next row if the first one is full (like with col-12). To me (and atleast the other person that had previously created an issue) the behavior of .col is not clear, and i would like to specify this in the docs.
The codepen in the docs actually does wrap, which makes this issue even more confusing (also shown in the codepen)
This is an issue that took me ~2 hours to clear up and i would like to enable others to avoid it in the future.

I see that col-grow is explained in the introduction, but it is (imo) easily overlooked, and could be mentioned again (especially in this context).

If you and the other maintainers do not consider this to be relevant for the documentation then we can stop discussing this. I found my solution. Otherwise, as i've said, i can create a PR for the documentation page to help avoid this pitfall.

**Comment by rstoenescu** [comment]
We'd appreciate if you give it a go with a PR.
